src/System.CommandLine/Binding/ModelBinder.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace System.CommandLine.Binding
+{
+    public class ModelBinder : BinderBase
+    {
+        public ModelDescriptor ModelDescriptor { get; }
+
+        public ModelBinder(ModelDescriptor modelDescriptor)
+        {
+            ModelDescriptor = modelDescriptor ?? throw new ArgumentNullException(nameof(modelDescriptor));
+        }
+
+        public ModelBinder(Type modelType) : this(ModelDescriptor.FromType(modelType))
+        {
+        }
+
+        public IReadOnlyCollection<BoundValue> GetConstructorArguments(BindingContext context)
+        {
+            return GetValues(context, ConstructorDescriptor.ParameterDescriptors);
+        }
+
+        public object CreateInstance(BindingContext context)
```

Comment:
Nit, consider GetInstance

Reply:
Can you explain how you think about the distinction here?

Reply:
On second look, Create is fine